test/katex-spec.js

```diff
@@ -2681,6 +2681,36 @@ describe("A macro expander", function() {
         expect("\\gdef{}{}").toNotParse();
     });
```

Comment:
It seems `\def`s expect a control sequence right after its command, so `\def{\foo}{1}` is invalid LaTeX.

Reply:
Also `\def\foo 1` doesn't work in TeX (the `1` is a pattern match, not an expansion). Unfortunately, I don't know how to detect either case. Our current MacroExpander doesn't distinguish between `\foo` and `{\foo}`, I believe (both return `[{text: '\foo'}]`).

Reply:
I'm pretty sure changing this would break a whole bunch of things, but maybe we should look at doing so sometime in the future.  We can add some kind of helper function to unwrap groups with a single child.

Reply:
Adding a TODO in `test/katex-spec.js`.